### DISCUSSION TITLE
Autostop on workspace upgrades

### DISCUSSION THREAD

**Discussion by samkeet** [coder/coder#15905] [General]
We recently updated out **Template** autostop timer from 24 hrs to 7 hours.

<img width="1453" alt="Screenshot 2024-12-17 at 2 56 22 PM" src="https://github.com/user-attachments/assets/f076b9a1-f1bd-4a9c-b9df-5dfaabd19fc9" />

But when I check some of the old workspaces, on previous versions of template before this change. The auto-stop time under `Workspace` settings still shows as 24 hours, despite being on the newest template version

<img width="1420" alt="Screenshot 2024-12-17 at 2 56 34 PM" src="https://github.com/user-attachments/assets/d564bccf-c2f1-46c9-a72e-ab096a086f0f" />

1. Does upgrading workspaces to latest template versions, not update auto-stop timer?
2. How can I propagate this change to workspaces before I made this change?
3. Is this really a paid feature in the new Coder versions as per https://coder.com/pricing#compare-plans ?

Coder version - v2.0.2

**Comment by samkeet** [comment]
@matifali Can you help here?

**Reply by matifali** [reply]
HI, @samkeet. Yes, enforcing auto/stop auto start is a paid feature.
In the OSS version, each developer can set their own auto-start and auto-stop schedule in their workspace schedule settings. 

The change takes effect after the workspace is restarted.
